Add tests for invite limits.

showell opened the pull request:
@gnprice FYI.

gnprice commented on zerver/lib/actions.py:
```diff
@@ -4435,16 +4435,26 @@ def check_invite_limit(user: UserProfile, num_invitees: int) -> None:
-                    raise InvitationError(msg, [], sent_invitations=False)
+        if user.realm.date_created - timezone_now() > newrealm_age:
+            # If the user's realm is sufficiently old, then we don't
+            # apply invite limits.  This code is targeted at the specific
```
We do still apply invite limits -- see the `raise` above. Which is important. :-)

Ditto on the "invite limits" in the next comment.

I think potentially a way to help make this logic clearer would be to pull this whole stanza (from `INVITES_DEFAULT_REALM_DAILY_MAX` to the end of the function) into a separate function, with a name like maybe `check_aggregate_invite_limit`. Then also these two comments can say "this limit" or "the aggregate invite limit".

gnprice replied:
Also possibly helpful for clarity: that function doesn't need `user` as a parameter, only `realm=user.realm`.

Actually, I guess the same is true of this `check_invite_limit` as a whole -- it's all about applying limits at the realm level and up.

gnprice replied:
Another idea for clarifying this: make the `if settings.OPEN_REALM_CREATION` at the top an early-return too. This whole function is for when the server allows anyone on the network to create a realm, because it's for spam where the realm itself belongs to the spammer.

gnprice commented on zerver/lib/actions.py:
```diff
@@ -4435,16 +4435,26 @@ def check_invite_limit(user: UserProfile, num_invitees: int) -> None:
+            # scenario of somebody using a new realm to launch spam.
+            return
+
+        if user.realm.max_invites >= default_max:
```
Logic changed here too. The `==` case is the most common one, so the behavior there is essential.

showell commented:
I addressed these:

* put back age check to date comparison (instead of age comparison)
* fixed == case of flipping inequalities
* passed in realm instead of user
* early-exit for settings.OPEN_REALM_CREATION

@gnprice I think this addresses all of your comments except splitting out a new function, which I agree with but will save for a subsequent commit.

timabbott commented:
Merged, thanks @showell!
